qiskit/transpiler/preset_passmanagers/level2.py

```diff
@@ -262,12 +262,15 @@ def _unroll_condition(property_set):
         sched = plugin_manager.get_passmanager_stage(
             "scheduling", scheduling_method, pass_manager_config, optimization_level=2
         )
+    init = common.generate_error_on_control_flow(
+        "The optimizations in optimization_level=2 do not yet support control flow."
+    )
```

Comment:
We technically could get a level2 pm that given the assumptions made in `common.py` is runnable with control flow. But it'd basically be `layout_method=dense`, `routing_method=stochastic`, `optimization=plugin`. It seems to be fairly unlikely in practice so this is fine I think.

Reply:
I was treating us as "supporting" an optimisation level if the defaults will work - for O2, you have to override everything that actually makes it O2 back to what's in O1, so I think it's better to reject `optimization_level=2` out-of-hand.

Reply:
I guess at O1, now that routing and layout are both Sabre by default there's an argument that we're kind of just doing that internally too, I suppose.  I was still thinking in terms of the old O1 that had dense/stochastic as its default.

Reply:
I'm fine with this for now, to a certain degree if people want to get more complicated maybe they should just be using custom pass managers. We just will have to document clearly in the release note that `optimization_level` 2 and 3 will error regardless of options with a control flow circuit.